hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/realtime/RealtimeCompactedRecordReader.java

```diff
@@ -55,13 +55,20 @@
   private final Set<String> deltaRecordKeys;
   private int recordKeyIndex = HoodieInputFormatUtils.HOODIE_RECORD_KEY_COL_POS;
   private Iterator<String> deltaItr;
+  private boolean logFileOnlySplit;
+  private HoodieMergedLogReader logReader;
 
   public RealtimeCompactedRecordReader(RealtimeSplit split, JobConf job,
       RecordReader<NullWritable, ArrayWritable> realReader) throws IOException {
     super(split, job);
     this.parquetReader = realReader;
-    this.deltaRecordMap = getMergedLogRecordScanner().getRecords();
+    HoodieMergedLogRecordScanner hoodieMergedLogRecordScanner = getMergedLogRecordScanner();
+    this.deltaRecordMap = hoodieMergedLogRecordScanner.getRecords();
     this.deltaRecordKeys = new HashSet<>(this.deltaRecordMap.keySet());
+    if (split.getPath().toString().contains(".log.")) {
+      this.logFileOnlySplit = true;
```

Comment:
Use `FSUtils.isLogFile` ?

Reply:
Keep `logReader` as `Option< HoodieMergedLogReader >` then we can eliminate the flag `logFileOnlySplit`.

Reply:
agree